### PR TITLE
Collapse transport versions for 8.16

### PR DESCRIPTION
**Pull request by thecoop**
Collapse transport versions into 8.16

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/TransportVersions.java [hunk 1]
```diff
@@ -104,78 +104,7 @@ static TransportVersion def(int id) {
     public static final TransportVersion V_8_14_0 = def(8_636_00_1);
     public static final TransportVersion V_8_15_0 = def(8_702_00_2);
     public static final TransportVersion V_8_15_2 = def(8_702_00_3);
-    public static final TransportVersion QUERY_RULES_LIST_INCLUDES_TYPES_BACKPORT_8_15 = def(8_702_00_4);
```
Comment (thecoop) [inline server/src/main/java/org/elasticsearch/TransportVersions.java:107]: @kderusso @markjhoy There are no more 8.15 releases planned, so this backport can just be removed here

File: server/src/main/java/org/elasticsearch/common/io/stream/StreamInput.java [hunk 2]
```diff
@@ -908,11 +908,8 @@ public final Instant readOptionalInstant() throws IOException {
     private ZonedDateTime readZonedDateTime() throws IOException {
         final String timeZoneId = readString();
         final Instant instant;
-        if (getTransportVersion().onOrAfter(TransportVersions.ZDT_NANOS_SUPPORT_BROKEN)) {
-            // epoch seconds can be negative, but it was incorrectly first written as vlong
-            boolean zlong = getTransportVersion().onOrAfter(TransportVersions.ZDT_NANOS_SUPPORT);
-            long seconds = zlong ? readZLong() : readVLong();
-            instant = Instant.ofEpochSecond(seconds, readInt());
+        if (getTransportVersion().onOrAfter(TransportVersions.V_8_16_0)) {
+            instant = Instant.ofEpochSecond(readZLong(), readInt());
```
Comment (thecoop) [inline server/src/main/java/org/elasticsearch/common/io/stream/StreamInput.java:912]: The support for broken zdt nanos can be removed here, both versions collapse into 8.16

File: server/src/main/java/org/elasticsearch/TransportVersions.java (hunk 1 above)
Comment (kderusso) [reply server/src/main/java/org/elasticsearch/TransportVersions.java:107]: Thanks for cleaning that up!

**Review by prdoyle — APPROVED** [review]

**Comment by thecoop** [comment]
BWC failures are also present on main

**Pull request merged**